### PR TITLE
Added quasar chat to detail view

### PR DESCRIPTION
**Pull request by poradz**
image

### CODE REVIEW

**Comment by mariuszkrzaczkowski** [comment]
## Migration Recommended

It appears you are still using the CoPilot OAuth App - we recommend you [migrate to the GitHub App](https://github.com/apps/blackduck-copilot/installations/new/permissions?suggested_target_id=10201412&repository_ids[]=28060246)

**Comment by mariuszkrzaczkowski** [comment]
## Black Duck Security Report

Merging #11216 into developer will not change security risk.

[Click here to see full report](https://copilot.blackducksoftware.com/github/repos/YetiForceCompany/YetiForceCRM/pull-requests/11216)

File: public_html/src/layouts/basic/modules/Chat/components/UpdateWatcher.vue [hunk 1]
```diff
@@ -0,0 +1,83 @@
+<!-- /* {[The file is published on the basis of YetiForce Public License 3.0 that can be found in the following directory: licenses/LicenseEN.txt or yetiforce.com]} */ -->
+<template></template>
+<script>
+import { createNamespacedHelpers } from 'vuex'
+const { mapGetters, mapActions, mapMutations } = createNamespacedHelpers('Chat')
+export default {
```
Comment (rskrzypczak) [inline public_html/src/layouts/basic/modules/Chat/components/UpdateWatcher.vue]: docBlock everywhere

File: modules/Chat/views/Detail.php [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+		parent::__construct();
+		$this->exposeMethod('showChatTab');
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function checkPermission(App\Request $request)
+	{
+		$currentUserPriviligesModel = Users_Privileges_Model::getCurrentUserPrivilegesModel();
```
Comment (rskrzypczak) [inline modules/Chat/views/Detail.php]: if (\App\User::getCurrentUserId() !== \App\User::getCurrentUserRealId()) {
			throw new \App\Exceptions\NoPermitted('ERR_NOT_ACCESSIBLE', 406);
		}
if (!$request->has('recordId') || Vtiger_Record_Model::getInstanceById($request->getInteger('recordId'))) {
			$this->recordModel = Vtiger_Record_Model::getInstanceById($request->getInteger('recordId')->isViewable());

				throw new \App\Exceptions\NoPermittedToRecord('ERR_NO_PERMISSIONS_FOR_THE_RECORD', 
		}

File: modules/Vtiger/views/Detail.php [hunk 3]
```diff
@@ -1074,14 +1074,19 @@ public function showSocialMedia(App\Request $request)
...
-	 *
-	 * @return \html
 	 */
 	public function showChat(App\Request $request)
 	{
-		return (new Chat_Detail_View())->showChatTab($request);
+		if (\App\User::getCurrentUserId() !== \App\User::getCurrentUserRealId()) {
+			throw new \App\Exceptions\NoPermitted('ERR_NOT_ACCESSIBLE', 406);
+		}
+		if (!$request->has('record') || !Vtiger_Record_Model::getInstanceById($request->getInteger('record'))->isViewable()) {
```
Comment (rskrzypczak) [inline modules/Vtiger/views/Detail.php]: remove

File: modules/Chat/actions/ChatAjax.php [hunk 4]
```diff
@@ -105,15 +107,60 @@ public function getMessages(App\Request $request)
...
+	 * @throws \App\Exceptions\IllegalValue
+	 */
+	public function getRoomsMessages(App\Request $request)
+	{
+		$rooms = $request->getArray('rooms');
+		$result = [];
+		$roomList = \App\Chat::getRoomsByUser();
+		$areNewEntries = false;
+
+		foreach ($rooms as $room) {
```
Comment (mariuszkrzaczkowski) [inline modules/Chat/actions/ChatAjax.php]:
```suggestion
		foreach ($request->getArray('rooms') as $room) {
```

File: modules/Chat/actions/ChatAjax.php [hunk 5]
```diff
@@ -105,15 +107,60 @@ public function getMessages(App\Request $request)
...
+			$recordId = $room['recordid'];
+			$roomType = $room['roomType'];
+			$chat = \App\Chat::getInstance($roomType, $recordId);
+			if (!$chat->isRoomExists()) {
+				return;
```
Comment (mariuszkrzaczkowski) [inline modules/Chat/actions/ChatAjax.php]: add error messages

**Pull request merged**